Automattic/woocommerce-payments
# refactor: rename payment method names in settings

Fixes https://github.com/Automattic/woocommerce-payments/issues/1869
Fixes https://github.com/Automattic/woocommerce-payments/issues/2030

#### Changes proposed in this Pull Request









I am changing the names of the payment methods used in the settings page closer to the ones used by Stripe.
This is a list of payment method names, for accuracy: https://stripe.com/docs/api/payment_methods/create


#### Testing instructions





This PR deprecates the `enabled_payment_method_ids` setting name.
This is because the `woocommerce_woocommerce_payments_settings.enabled_payment_method_ids` setting (which has been used during development for the WCPay settings redesign) _could_ contain an invalid value (`woocommerce_payments`). You might have had that value set while developing.
For this reason, I renamed it to `upe_enabled_payment_method_ids`, which just generates less issues than renaming the old setting used for development.

**PREFACE: Ensure you have the latest version of WCPay Dev tools**
```
cd docker/wordpress/wp-content/plugins/woocommerce-payments-dev-tools && git pull
```

**User migrating to the new settings**
- In WCPay dev, ensure you have the "Enable grouped settings", "Enable UPE checkout" and "Add UPE additional payment methods" features disabled
- Go to the WCPay settings page: http://localhost:8082/wp-admin/admin.php?page=wc-settings&tab=checkout&section=woocommerce_payments
- Change/save some settings
- In WCPay dev, enable the "Enable grouped settings" feature
- Go to the settings page: http://localhost:8082/wp-admin/admin.php?page=wc-settings&tab=checkout&section=woocommerce_payments
- Settings are now updated with the new design
- Saving settings still works
![Screen Shot 2021-06-08 at 12 59 37 PM](https://user-images.githubusercontent.com/273592/121234532-6c53ae00-c859-11eb-92bc-d8bf56ed1c79.png)

**User with UPE**
- In WCPay dev, enable the "Enable grouped settings", "Enable UPE checkout" and "Add UPE additional payment methods" features
- Go to the settings page: http://localhost:8082/wp-admin/admin.php?page=wc-settings&tab=checkout&section=woocommerce_payments
- New section has been added, which allows to add UPE payment methods
- New payment methods can be added/saved and are persisted on page refresh
![Screen Shot 2021-06-08 at 1 01 46 PM](https://user-images.githubusercontent.com/273592/121234841-c6ed0a00-c859-11eb-9d05-684a5989aa11.png)

-------------------
